### PR TITLE
Encounters list page: API logics, search, delete modal, infinite scroll with pagination

### LINKED ISSUE #65: Create Encounters page
## Description
 
The Encounters page displays all encounters the user saved, wireframe example as below:
Screen Shot 2022-03-08 at 6 58 40 PM
Screen Shot 2022-03-11 at 5 38 46 PM

## Acceptance Criteria

- [ ] Displays a search bar, search results should reflect on the page and replace the original list, similar to the behaviour on the Persons page.
Screen Shot 2022-03-11 at 5 31 34 PM
- [ ] Displays a 'New Encounter' button, onClick lead users to Create Persons page
Screen Shot 2022-03-11 at 5 40 31 PM
- [ ] Displays a list (scrollable) of encounters with the first three lines of details, person(s) with profile image, date we met, where we met and a delete button, delete button should trigger a pop up Modal component similar to this example below with description "Are you sure you want to delete this encounter? You cannot undo this action."
Screen Shot 2022-03-11 at 5 34 05 PM
- [ ] Clicking on one of the encounter cards should lead users to View Encounter Details page.
Screen Shot 2022-03-11 at 5 35 44 PM
- [ ] Hovering over one of the encounter cards should display an overview of encounter details on the right as shown on the above screenshots

## Implementation details

- Use CustomButton, EncounterCard, SearchBar components to implement this page

## Testing notes

- Please add tests to ensure all acceptance criteria are met before merging.

### PR DESCRIPTION
**Pull request by rainasong**
- Added API logics for getting encounters with pagination to get data from backend
- Infinite scroll works with pagination
- Implemented search API logic, dynamically render results
- Added delete logic, toasts and warning modal
- Added routing to new encounter page
- Added initial loading UI
- Fixed several styling issues

Fixes #65 

**Demo:**
Encounters List:
Screen Shot 2022-03-19 at 4 22 31 PM
When hover:
Screen Shot 2022-03-19 at 4 22 34 PM (2)
Search:
Screen Shot 2022-03-19 at 4 30 05 PM
Delete:
Screen Shot 2022-03-19 at 4 23 30 PM
Screen Shot 2022-03-19 at 4 23 37 PM

Pagination: (Page size set to 5 for demo purpose, normally should be 10)

https://user-images.githubusercontent.com/37087904/159104958-d23ba89c-ef3b-400e-b9d9-af61ffcddd11.mp4

### CODE REVIEW

**Review by lusicle — APPROVED** [review]
lgtm! 😃

**Comment by lin8231** [comment]
You may wanna put a line break in between your video link and the text above it so that the video embeds :)

File: forgettable-frontend/src/services/index.js [hunk 1]
```diff
@@ -112,7 +123,20 @@ export const deleteEncounter = async (id) => {
...
+ */
+export const searchEncounter = async (searchString) => {
+  return await getData('encounters/?term=' + searchString);
+};
 
+/**
+ * Searches Persons from the database given a search string
+ * @param {string} searchString string to search
+ * @return {Promise}
+ */
```
Comment (lin8231) [inline forgettable-frontend/src/services/index.js:139]: Thanks for adding this comment! I think I forgot to haha

File: forgettable-frontend/src/pages/EncountersListPage/EncountersListPage.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  };
+
+  const fetchMoreData = () => {
+    if (!hasMore) {
+      return;
+    }
+
+    setTimeout(async () => {
+      const newPageNum = pageNum + 1;
+      const newResult = await getEncountersByPage(newPageNum, LIMIT);
```
Comment (lin8231) [inline forgettable-frontend/src/pages/EncountersListPage/EncountersListPage.js]: Noticing that there isn't a LIMIT variable, maybe you forgot to rename either LIMIT or PAGE_SIZE?

**Review by lin8231 — CHANGES_REQUESTED** [review]
Found a tiny problem I think. Otherwise super good job!!!

File: forgettable-frontend/src/pages/EncountersListPage/EncountersListPage.js (hunk 2 above)
Comment (lin8231) [reply forgettable-frontend/src/pages/EncountersListPage/EncountersListPage.js]: Nevermind, you already got to it as I was reviewing haha

Comment (rainasong) [reply forgettable-frontend/src/pages/EncountersListPage/EncountersListPage.js]: Yea just updated it!

**Review by lin8231 — APPROVED** [review]
Thanks so much, amazing!

**Comment by lin8231** [comment]
Super excited to get this merged!

**Comment by rainasong** [comment]
> Super excited to get this merged!

Hehe our app finally gonna look more 'functional'😂

**Review by lusicle — APPROVED** [review]
lgtm! 😃

**Review by lin8231 — APPROVED** [review]
Reapproving after merge

**Pull request merged**